core/src/main/scala/kafka/coordinator/GroupCoordinator.scala

```diff
@@ -454,23 +454,45 @@ class GroupCoordinator(val brokerId: Int,
     delayedOffsetStore.foreach(groupManager.store)
   }
 
+  def getPartitions(groupId: String): List[TopicPartition] = {
+      val group = groupManager.getGroup(groupId)
+      group.fold(List[TopicPartition]())(_.allOffsets.keySet.toList)
+  }
 
   def handleFetchOffsets(groupId: String,
-                         partitions: Seq[TopicPartition]): Map[TopicPartition, OffsetFetchResponse.PartitionData] = {
+                         partitions: Seq[TopicPartition],
+                         apiVersion: Short): (Map[TopicPartition, OffsetFetchResponse.PartitionData], Short) = {
+    require(apiVersion >= 1)
+
     if (!isActive.get) {
-      partitions.map { topicPartition =>
-        (topicPartition, new OffsetFetchResponse.PartitionData(OffsetFetchResponse.INVALID_OFFSET, "", Errors.GROUP_COORDINATOR_NOT_AVAILABLE.code))}.toMap
+      (apiVersion match {
```

Comment:
Mainly what I'm trying to achieve is keeping version handling logic out of `GroupCoordinator` as much as possible. So what I had in mind is a constructor or a factory which accepts a top-level error code and a list of the partitions. In the case of the old version, we take the top-level error code and override the partition-level errors. In the case of the new version, we can ignore the partition data and just return the top-level error code.